pkg/search/postgres/store.go

```diff
@@ -180,3 +180,15 @@ func (s *GenericStore[T, PT]) Walk(ctx context.Context, fn func(obj PT) error) e
 	}
 	return nil
 }
+
+// GetAll retrieves all objects from the store.
+func (s *GenericStore[T, PT]) GetAll(ctx context.Context) ([]*T, error) {
+	defer s.setPostgresOperationDurationTime(time.Now(), ops.GetAll)
+
+	var objs []*T
```

Comment:
Why not `var objs []PT` and have `([]PT, error)` as return type ?

Reply:
Indeed this will save us a cast